sage_modabvar/abvar_newform.py

```diff
@@ -235,3 +238,77 @@ def _calculate_endomorphism_generators(self):
 
         return V.saturation().basis()
 
+    def frobenius_polynomial(self, p):
+        """
+        Computes the frobius polynomial at `p`.
+
+        INPUT:
+
+        - ``p`` -- prime number
+
+        OUTPUT:
+            a rational polynomial
+
+        EXAMPLES::
+            sage: f = Newform('39b','a')
+            sage: A=AbelianVariety(f)
+            sage: A.frobenius_polynomial(5)
+            x^4 + 2*x^2 + 25
+        """
+        f = self.__f
+        Kf = f.base_ring()
+        L = Kf.galois_closure('a')
+        emb = Kf.embeddings(L)[0]
+        eps = f.character()
+        ap = f.qexp(p+1).coefficients()[-1]
```

Comment:
This seems like a really dumb way to get the p-th coefficient.  What if p = 997?  You're doing a massive amount of wasted work getting all previous a_p, using them to get all a_n, then throwing all that away.